src/model/schema.js

```diff
@@ -465,7 +465,7 @@ export class MarkType extends SchemaItem {
 // with node type constructors and another similar object associating
 // mark names with mark type constructors.
 export class SchemaSpec {
-  // :: (?Object<NodeType>, ?Object<MarkType>)
+  // :: (?Object<SubClass<NodeType>>, ?Object<SubClass<MarkType>>)
```

Comment:
Yeah, it's very ad-hoc. Note though that the current type hint is wrong: This is not an object with `NodeType` instances as values, but an object with `NodeType` subclasses as values. Not `{ blockquote: new BlockQuote }`, but `{ blockquote: BlockQuote}`. There are probably better ways to solve this than introdocing an arbitrary pseudo-type, but I think you should fix it :)